--- a/usr/local/pnp4nagios/share/application/models/data.php    2021-02-19 17:21:11.482480107 +0000
+++ b/usr/local/pnp4nagios/share/application/models/data.php    2021-02-19 17:20:53.726195054 +0000
@@ -976,7 +976,7 @@
     */
     public function getFirstPage(){
         $pages = $this->getPages();
-        if(sizeof($pages) > 0 ){
+        if($pages && sizeof($pages) > 0 ){
             return urldecode($pages[0]);
         }else{
             return FALSE;
